src/math/trig.c

```diff
@@ -94,9 +122,201 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine,cosine*/
 	*th_cdc_fxp = th_rad_fxp;
-
+}
+/**
+ * Arguments	: int32_t sinvalue Q2.30
+ *		  int16_t numiters
+ *		  const int32_t LUT[30]
+ * Return Type	: int32_t
+ * Description	: Determines input value is scalar [logical 1 (true) logical 0 (false)]
+ */
+static int32_t iscalarcordicasin(int32_t sinvalue, int16_t numiters,
+				 const int32_t LUT[CORDIC_30B_ITABLE_SIZE])
+{
+	int32_t b_i;
+	int32_t b_sinvalue;
+	int32_t i;
+	int32_t x;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z;
+	int16_t j;
+	int16_t k;
+	b_sinvalue = (sinvalue >> 1);
+	if (b_sinvalue > cord_arcsincos_q28fl) {
+		x = 0;
+		y = cord_arcsincos_q30fl;
+		z = cord_arcsincos_piovertwo_q29fl;
+	} else {
+		x = cord_arcsincos_q30fl;
+		y = 0;
+		z = 0;
+	}
+	b_sinvalue <<= 1;
+	i = (int32_t)((int16_t)(numiters - 1));
+	for (b_i = 0; b_i < i; b_i++) {
+		j = (int16_t)((b_i + 1) << 1);
```

Comment:
can we comment this function too, 
btw - we always do this calculation but the result is always discarded ```if ((j > 31)```, would be better to only do the calculation if b_i < 15 ?

Reply:
|b_i		|15	   |
	|b_sinValue	|1455896920|
	|i		|30	   |
	|j		|32	   |
	|k		|14	   |